exp/topology/desiredstate/desired_state.go

```diff
@@ -526,6 +527,11 @@ func (g *generator) computeControlPlaneVersion(ctx context.Context, s *scope.Sco
 	}
 
 	if feature.Gates.Enabled(feature.RuntimeSDK) {
```

Comment:
I think we should extend the Runtime SDK e2e test to cover this, like the other BeforeClusterUpgrade hook

(let me know if you want to chat about how to do that)